### DISCUSSION TITLE
Replacing mainboard

### DISCUSSION THREAD

**Discussion by Altduckthing** [bassamanator/Sovol-SV06-firmware#31] [Q&A]
I'm a little scared to flash the firmware to sv06 as I had to get a replacement last time I did 
So would it be possible to use this 
https://www.amazon.com/Sovol-32-bit-Silent-Motherboard-Mainboard/dp/B0BTRZN3H7

**Comment by bassamanator** [comment]
Looks like the standard board. Be sure to use an appropriate microSD. You can test the waters by flashing a newer marlin firmware.

From what I've seen, printer boards in general are hard to brick. You can usually recover by using `avrdude`. Tell me what happened with that board?

**Reply by Altduckthing** [reply]
I had no idea about that command. I tried to flash the new firmware with the standard mircoSD and it bricked. I looked for solutions online and couldn't find any that would work with the standard mircosd card. At that point I returned the whole printer to Amazon and bought a new one. This is really me trying to prevent something like that happening again and I also bought a new sd card just in case.

**Reply by bassamanator** [reply]
You need not worry. I think the only way to brock this board is to cut off power during a flash, and then again it's probably recoverable via `avrdude`.

In any case, please close this discussion, and good luck!

**Reply by Altduckthing** [reply]
Could you point me to a avrdude tutorial, I might try this on the stock board then.

**Reply by bassamanator** [reply]
For such a util, it's more about reading the docs: https://github.com/avrdudes/avrdude

On my Mega S, I recovered many times via `avrdude -p m2560 -P /dev/ttyUSB0 -c wiring -D -U flash:w:MEGA_S_DGUS_v1.2.0-beta.3.hex`, I'm confident that these particular flags will not work for this board.

```
$ avrdude --help
Usage: avrdude [options]
Options:
  -p <partno>                Specify AVR device
  -p <wildcard>/<flags>      Run developer options for matched AVR devices
  -b <baudrate>              Override RS-232 baud rate
  -B <bitclock>              Specify bit clock period (us)
  -C <config-file>           Specify location of configuration file
  -c <programmer>            Specify programmer type
  -c <wildcard>/<flags>      Run developer options for matched programmers
  -A                         Disable trailing-0xff removal from file and AVR read
  -D                         Disable auto erase for flash memory; implies -A
  -i <delay>                 ISP Clock Delay [in microseconds]
  -P <port>                  Specify connection port
  -F                         Override invalid signature or initialisation check
  -e                         Perform a chip erase
  -O                         Perform RC oscillator calibration (see AVR053)
  -U <memtype>:r|w|v:<filename>[:format]
                             Memory operation specification
                             Multiple -U options are allowed, each request
                             is performed in the order specified
  -n                         Do not write anything to the device
  -V                         Do not verify
  -t                         Enter terminal mode
  -E <exitspec>[,<exitspec>] List programmer exit specifications
  -x <extended_param>        Pass <extended_param> to programmer
  -v                         Verbose output; -v -v for more
  -q                         Quell progress output; -q -q for less
  -l logfile                 Use logfile rather than stderr for diagnostics
  -?                         Display this usage
```
If you figure this out please post here so that others might benefit.

Also, you need not go this route, simply attempt to flash original firmware back using the flashing method found on this repo.

**Reply by Altduckthing** [reply]
OK I was able to install the new marlin firmware onto my sv06 and I will try to install klippet later on but I have a quick question about this https://www.youtube.com/watch?v=lxdEzXS6AMU would this work on klipper I know some parts of the gcode don't work on klipper so I was hoping you could check it out

**Reply by bassamanator** [reply]
Never used it, so I'm unsure. Ask the author, he's a redditor.
